src/NLog/Targets/WebServiceTarget.cs

```diff
@@ -312,135 +335,302 @@ private Uri BuildWebServiceUrl(object[] parameterValues)
+        private class HttpPostJsonFormatter : HttpPostTextFormatterBase
+        {
+            public HttpPostJsonFormatter(WebServiceTarget target) : base(target)
+            { }
+
+            protected override string ContentType
+            {
+                get { return "application/json"; }
+            }
+
+            protected override string Separator
+            {
+                get { return ","; }
+            }
+
+            protected override string GetFormattedContent(string parametersContent)
+            {
+                return string.Format("{{{0}}}", parametersContent);
+            }
+
+            protected override string GetFormattedParameter(MethodCallParameter parameter, object value)
+            {
+                return string.Format("\"{0}\":{1}",
+                    parameter.Name,
+                    GetJsonValueString(value));
+            }
+
+            private string GetJsonValueString(object value)
+            {
+                return ConfigurationItemFactory.Default.JsonSerializer.SerializeObject(value);
+            }
+        }
+
+        private class HttpPostSoap11Formatter : HttpPostSoapFormatterBase
         {
-            request.Method = "POST";
-            request.ContentType = "text/xml; charset=" + this.Encoding.WebName;
+            public HttpPostSoap11Formatter(WebServiceTarget target) : base(target)
+            {
+            }
 
-            var ms = new MemoryStream();
-            XmlWriter xtw = XmlWriter.Create(ms, new XmlWriterSettings { Encoding = this.Encoding });
+            protected override string SoapEnvelopeNamespace
+            {
+                get { return WebServiceTarget.SoapEnvelopeNamespace; }
+            }
 
-            xtw.WriteStartElement(soapname, "Envelope", soapEnvelopeNamespace);
-            xtw.WriteStartElement("Body", soapEnvelopeNamespace);
-            xtw.WriteStartElement(this.MethodName, this.Namespace);
-            int i = 0;
-            foreach (MethodCallParameter par in this.Parameters)
+            protected override string SoapName
             {
-                xtw.WriteElementString(par.Name, Convert.ToString(parameterValues[i], CultureInfo.InvariantCulture));
-                i++;
+                get { return "soap"; }
             }
 
-            xtw.WriteEndElement(); // methodname
-            xtw.WriteEndElement(); // Body
-            xtw.WriteEndElement(); // soap:Envelope
-            xtw.Flush();
+            protected override void InitRequest(HttpWebRequest request)
+            {
+                base.InitRequest(request);
+
+                string soapAction;
+                if (Target.Namespace.EndsWith("/", StringComparison.Ordinal))
+                {
+                    soapAction = Target.Namespace + Target.MethodName;
+                }
+                else
+                {
+                    soapAction = Target.Namespace + "/" + Target.MethodName;
```

Comment:
Consider changing into this (Maybe also use string.Concat for the one without slash for consistency)

soapAction = string.Concat(Target.Namespace, "/", Target.MethodName);